react-components/src/architecture/base/concreteCommands/pointCloud/PointCloudFilterCommand.test.ts

```diff
@@ -0,0 +1,111 @@
+import { assert, describe, expect, test, beforeEach } from 'vitest';
+import { PointCloudFilterCommand } from './PointCloudFilterCommand';
+import { isOdd } from '../../utilities/extensions/mathUtils';
+import { createFullRenderTargetMock } from '../../../../../tests/tests-utilities/fixtures/createFullRenderTargetMock';
+import { PointCloudDomainObject } from '../../../concrete/reveal/pointCloud/PointCloudDomainObject';
+import {
+  createPointCloudMock,
+  pointCloudClasses
+} from '../../../../../tests/tests-utilities/fixtures/pointCloud';
+import { count } from '../../utilities/extensions/arrayUtils';
+
+describe(PointCloudFilterCommand.name, () => {
+  let command: PointCloudFilterCommand;
+  let domainObject: PointCloudDomainObject;
+
+  beforeEach(() => {
+    const renderTarget = createFullRenderTargetMock();
+    domainObject = new PointCloudDomainObject(createPointCloudMock());
+    renderTarget.rootDomainObject.addChildInteractive(domainObject);
+
+    command = new PointCloudFilterCommand();
+    command.attach(renderTarget);
+    command.initializeChildrenIfNeeded();
```

Comment:
I must admit I'm very against this kind of pattern - it seems unnecessary that we must call two additional methods after the constructor before the command is usable